File: src/types/sentiment.ts
export type SentimentLabel =
  | "strong-positive"
  | "positive"
  | "neutral"
  | "negative"
  | "strong-negative";

export interface SentimentSplit {
  bullish: number;
  neutral: number;
  bearish: number;
}

export type SentimentSourceType = "news" | "commentary" | "social" | "macro";

export interface SentimentSource {
  id: string;
  type: SentimentSourceType;
  label: string;
  sentiment: SentimentLabel;
  confidence: number;
  contributionPercent: number;
  updatedAt: string;
}

export interface SentimentTimelinePoint {
  time: string;
  label: SentimentLabel;
  score: number;
}

export interface SymbolSentiment {
  symbol: string;
  split: SentimentSplit;
  sources: SentimentSource[];
  timeline: SentimentTimelinePoint[];
}

File: backend/app/schemas/sentiment.py
from datetime import datetime
from typing import Literal

from app.schemas.market import CamelModel

SentimentLabel = Literal["strong-positive", "positive", "neutral", "negative", "strong-negative"]
SentimentSourceType = Literal["news", "commentary", "social", "macro"]


class SentimentSplitSchema(CamelModel):
    bullish: int
    neutral: int
    bearish: int


class SentimentSourceSchema(CamelModel):
    id: str
    type: SentimentSourceType
    label: str
    sentiment: SentimentLabel
    confidence: int
    contribution_percent: int
    updated_at: datetime


class SentimentTimelinePointSchema(CamelModel):
    time: str
    label: SentimentLabel
    score: int


class SymbolSentimentSchema(CamelModel):
    symbol: str
    split: SentimentSplitSchema
    sources: list[SentimentSourceSchema]
    timeline: list[SentimentTimelinePointSchema]
